test/test_integration_list.py

```diff
@@ -0,0 +1,324 @@
+                          f'    "{self.client_params.account}:variable:one/password",\n'
+                          f'    "{self.client_params.account}:webservice:somewebservice"\n]\n')
+
+    # TODO This will need to be changed when UX is included
+    @integration_test
+    def test_list_help_returns_help_screen(self):
+        self.setup_cli_params({})
+        output = self.invoke_cli(self.cli_auth_params, ['list', '-h'])
+        self.assertIn("usage:  list", output)
+
+    @integration_test
+    def test_list_inspect_user_returns_info_on_user(self):
+        self.setup_cli_params({})
+        self.invoke_cli(self.cli_auth_params,
+               ['policy', 'replace', 'root', self.environment.path_provider.get_policy_path("conjur")])
+        output = self.invoke_cli(self.cli_auth_params, ['list', '--inspect', '--kind', 'user'])
+        self.assertIn(f'        "id": "{self.client_params.account}:user:someuser",\n'
+                      f'        "owner": "{self.client_params.account}:user:admin",\n'
+                      f'        "policy": "{self.client_params.account}:policy:root",\n'
+                      f'        "permissions": [],\n'
+                      f'        "annotations": [],\n'
+                      f'        "restricted_to": []\n'
+                      f'    }}\n'
+                      f']\n', output)
+
+    @integration_test
+    def test_list_kind_layer_returns_layers(self):
+        self.setup_cli_params({})
+        with self.assertLogs('', level='DEBUG') as mock_log:
+            output = self.invoke_cli(self.cli_auth_params, ['list', '-k', 'layer'])
+            self.assertEquals(output,
+                              f'[\n    "{self.client_params.account}:layer:somelayer"\n]\n')
+            self.assertIn("Executing list command with the following constraints: {'kind': 'layer'}",
+                          str(mock_log.output))
+
+    @integration_test
+    def test_list_kind_user_returns_users(self):
+        self.setup_cli_params({})
+        output = self.invoke_cli(self.cli_auth_params, ['list', '-k', 'user'])
+        self.assertEquals(output,
+                          f'[\n    "{self.client_params.account}:user:someuser"\n]\n')
+
+    @integration_test
+    def test_list_kind_nonexistent_returns_empty_list(self):
+        self.setup_cli_params({})
+        output = self.invoke_cli(self.cli_auth_params, ['list', '-k', 'nonexistentkind'])
+        self.assertEquals(output,
+                          f'[]\n')
+
+    @integration_test
+    def test_list_kind_variable_returns_variables(self):
+        self.setup_cli_params({})
+        output = self.invoke_cli(self.cli_auth_params, ['list', '-k', 'variable'])
+        self.assertEquals(output,
+                          f'[\n    "{self.client_params.account}:variable:one/password"\n]\n')
+
+    @integration_test
+    def test_list_kind_group_returns_groups(self):
+        self.setup_cli_params({})
```

Comment:
if this is calls for every method why not in the constructor?

Reply:
I removed from all test definitions and moved to setup so it appears only once